pypgatk/ensembl/data_downloader.py

```diff
@@ -16,326 +16,367 @@
+            total_files.extend(files)
+    else:
+      for species in self._ensembl_species:
+        self.get_logger().debug(
+          "Downloading the data for the specie -- " + species[self.CONFIG_REST_API_TAXON_ID])
+        if not self.get_pipeline_parameters()[self.CONFIG_KEY_SKIP_PROTEIN]:
+          prot_files = self.get_pep_files(species, grch37)
+          files.extend(prot_files)
+        if not self.get_pipeline_parameters()[self.CONFIG_KEY_SKIP_GTF]:
+          gtf_files = self.get_gtf_files(species, grch37)
+          files.extend(gtf_files)
+        if not self.get_pipeline_parameters()[self.CONFIG_KEY_SKIP_CDS]:
+          cds_files = self.get_cds_files(species, grch37)
+          files.extend(cds_files)
+        if not self.get_pipeline_parameters()[self.CONFIG_KEY_SKIP_CDNA]:
+          cdna_files = self.get_cdna_files(species, grch37)
+          files.extend(cdna_files)
+        if not self.get_pipeline_parameters()[self.CONFIG_KEY_SKIP_NCRNA]:
+          ncrna_files = self.get_ncrna_files(species, grch37)
+          files.extend(ncrna_files)
+        if not self.get_pipeline_parameters()[self.CONFIG_KEY_SKIP_VCF]:
+          vcf_files = self.get_vcf_files(species)
+          files.extend(vcf_files)
+
+        total_files.extend(files)
+        self.get_logger().debug("Files downloaded -- " + ",".join(files))
+        total_files.extend(files)
+
+    return total_files
+
+  def get_cds_files(self, species: dict, grch37=False) -> list:
+    """
         Get the cds files for a specific species object.
         :return: List of files names.
         """
-        files = []
-        try:
-            file_name = '{}.{}.cds.all.fa.gz'.format(species['name'][0].upper() + species['name'][1:],
-                                                     species['assembly'])
-            file_url = '{}/release-{}/fasta/{}/cds/{}'.format(
-                self.get_default_parameters()[self.CONFIG_KEY_DATA_DOWNLOADER][self.CONFIG_KEY_ENSEMBL_FTP][
-                    self.CONFIG_KEY_BASE_URL],
-                species['release'], species['name'], file_name)
-            files.append(download_file(file_url, self.get_local_path_root_ensembl_repo() + '/' + file_name, self.get_logger()))
-        except KeyError:
-            print("No valid info is available species: ", species)
-
-        return files
-
-    def get_cdna_files(self, species: dict) -> list:
-        """
+    files = []
+    try:
+      if grch37:
+        species['assembly'] = 'GRCh37'
+      file_name = '{}.{}.cds.all.fa.gz'.format(species['name'][0].upper() + species['name'][1:],
+                                               species['assembly'])
+      file_url = '{}/release-{}/fasta/{}/cds/{}'.format(
+        self.get_default_parameters()[self.CONFIG_KEY_DATA_DOWNLOADER][self.CONFIG_KEY_ENSEMBL_FTP][
+          self.CONFIG_KEY_BASE_URL], species['release'], species['name'], file_name)
+      if grch37:
+        file_url = '{}/grch37/release-{}/fasta/{}/cds/{}'.format(
```

Comment:
release number should be a user parameter, if not given then get it from species['release']

Reply:
Would be good to have the releases previous to 38, the problem is that ENSEMBL do not give that information with the same structure. Lets wait until a user ask for the feature in the future.